go/host/host.go

```diff
@@ -1115,3 +1125,22 @@ func (h *host) catchUpL1Block() bool {
 	}
 	return true
 }
+
+func (h *host) RegisterService(name string, service hostcommon.Service) {
+	if _, ok := h.services[name]; ok {
+		h.logger.Crit("service already registered", "name", name)
+	}
+	h.services[name] = service
+}
+
+func (h *host) GetService(name string) hostcommon.Service {
+	service, ok := h.services[name]
+	if !ok {
+		h.logger.Crit("requested service not registered", "name", name)
+	}
+	return service
+}
+
+func (h *host) l1Repo() hostcommon.L1BlockRepository {
```

Comment:
These lookups don't need to be used all over the place (I'm hoping we won't have the lookups at all).

When the enclave-guardian is created for example it will have the l1Repo injected into it. It's just used in here atm because the logic hasn't moved out yet.